File: examples/bot.py
# Connect to Slack

import quark_twisted_runtime
import slack


class Handler(slack.SlackHandler):

    def onHello(self, greeting):
        print "Hello received:", greeting

    def onSlackError(self, error):
        print "Error:", error.code, error.text

    def onSlackEvent(self, event):
        print "Event:", event

    def onMessage(self, message):
        print "Received:", message.text
        if message.text and "quark" in message.text:
            message.channel.send("bot.py is excited about \"%s\"" % message.text.replace("quark", "[bleep]"))


def main():
    # you need to go to https://api.slack.com/web and generate an access token
    token = "xoxp-3910618990-4153681316-13601165377-914ddc6f19"

    runtime = quark_twisted_runtime.get_twisted_runtime()
    client = slack.Client(runtime, token, Handler())
    client.connect()
    runtime.launch()


if __name__ == '__main__':
    main()

File: examples/bot.js
var runtime = require("quark_node_runtime.js")
var slack = require("slack")
var util = require("util")

function Handler() {
    this.onHello = function(hello) {
        console.log(hello)
    }
    this.onSlackError = function(error) {
        console.log(error)
    }
    this.onSlackEvent = function(event) {
        console.log(event)
    }
    this.onMessage = function(message) {
        console.log(message)
        if (message.text != null && message.text.indexOf("quark") > -1) {
            message.channel.send("quarkbot: " + message.text)
        }
    }
}

util.inherits(Handler, slack.SlackHandler)

// you need to go to https://api.slack.com/web and generate an access token
var token = "xoxp-3910618990-4153681316-13601165377-914ddc6f19"
client = new slack.Client(runtime, token, new Handler());
client.connect();
